Samples/NetworkMediaStreamer/RTSPServer/src/RTSPServer.cpp

```diff
@@ -0,0 +1,169 @@
+// Copyright (C) Microsoft Corporation. All rights reserved.
+
+#pragma comment(lib, "Ws2_32.lib")
+#include <pch.h>
+
+STDMETHODIMP RTSPServer::StopServer()
+{
+    HRESULT_EXCEPTION_BOUNDARY_START;
+    auto apiLock = std::lock_guard(m_apiGuard);
+    if (m_callbackHandle)
+    {
+        UnregisterWait(m_callbackHandle.detach());
+    }
+    WSACloseEvent(m_acceptEvent.detach());
+
+    for (auto& session : m_Sessions)

```

Comment:
>m_Sessions [](start = 25, length = 10)

Ideally this should use uniquePtr, shared pointer, or stack memory (only if no VTable) then memory management is auto handled for you

Reply:
i will try to make this uniquePtr

---
In reply to: [491107434](https://github.com/microsoft/Windows-Camera/pull/13#discussion_r491107434) [](ancestors = 491107434)